sympy/functions/elementary/trigonometric.py

```diff
@@ -1484,6 +1497,10 @@ def _eval_rewrite_as_sqrt(self, arg, **kwargs):
             return None
         return y
 
+    def _eval_rewrite_as_besselj(self, arg, **kwargs):
+        from sympy.functions.special.bessel import besselj
+        return Piecewise(((sqrt(2)*sqrt(pi)*sqrt(arg)*besselj(-S.Half, arg)/2)/(sqrt(2)*sqrt(pi)*sqrt(arg)*besselj(S.Half, arg)/2), Ne(arg, 0)), (zoo, True))
```

Comment:
I'm not sure if we should expect Piecewise to make special cases for zoo like this...

Reply:
Then what shall we do ?

Reply:
The Piecewise is not needed here because 
```
In [7]: arg = x

In [8]: e = (sqrt(2)*sqrt(pi)*sqrt(arg)*besselj(-S.Half, arg)/2)/(sqrt(2)*sqrt(pi)*sqrt(arg)*besselj(S.Ha
   ...: lf, arg)/2)

In [9]: e
Out[9]: 
besselj(-1/2, x)
────────────────
besselj(1/2, x) 

In [10]: e.subs(x, 0)
Out[10]: zoo

```
Also we don't need the `sqrt(2)` etc because it will cancel out in the division.